### PR TITLE
Continue fs-extra migration

### PR DESCRIPTION
**Pull request by raphinesse**
I left the commits small so changes can be more easily tracked. Feel free to squash after changes have been reviewed. You can [add me as a Co-Author](https://help.github.com/articles/creating-a-commit-with-multiple-authors/).

### CODE REVIEW

File: spec/create.spec.js [hunk 1]
```diff
@@ -35,10 +35,9 @@ var project = path.join(tmpDir, appName);
 
 // Setup and teardown test dirs
 beforeEach(function () {
-    fs.removeSync(project);
-    fs.ensureDirSync(tmpDir);
+    fs.emptyDirSync(tmpDir);
 });
-afterEach(function () {
+afterAll(function () {
```
Comment (brody2consult) [inline spec/create.spec.js]: Why this change and why here?

File: index.js [hunk 2]
```diff
@@ -309,7 +309,7 @@ function copyTemplateFiles (templateDir, projectDir, isSubDir) {
     // if template is a www dir
     if (path.basename(templateDir) === 'www') {
         copyPath = path.resolve(templateDir);
-        fs.copySync(copyPath, projectDir);
+        fs.copySync(copyPath, path.resolve(projectDir, 'www'));
```
Comment (brody2consult) [inline index.js]: I am wondering why this problem does not show up when I run the spec?

File: index.js [hunk 3]
```diff
@@ -400,7 +400,7 @@ function linkFromTemplate (templateDir, projectDir) {
     // if template/www/config.xml then copy to project/config.xml
     copyDst = path.join(projectDir, 'config.xml');
     if (!fs.existsSync(copyDst) && fs.existsSync(copySrc)) {
-        fs.copySync(copySrc, projectDir);
```
Comment (brody2consult) [inline index.js]: Ditto: I am wondering why this problem does not show up when I run the spec?

File: spec/create.spec.js (hunk 1 above)
Comment (raphinesse) [reply spec/create.spec.js]: Why: Since with `emptyDirSync` we empty `tmpDir` in `beforeEach` so there's no need to delete the `tmpDir` itself after every test, only at the very end.

Why here: Because this change is kind of enabled by the rich `fs-extra` API :man_shrugging:

File: index.js [hunk 4]
```diff
@@ -228,8 +228,8 @@ module.exports = function (dir, optionalId, optionalName, cfg, extEvents) {
 
                 // If following were not copied/linked from template, copy from stock app hello world
                 // TODO: get stock package.json if template does not contain package.json;
-                copyContentsIfNotExists(stockAssetPath('www'), path.join(dir, 'www'));
-                copyContentsIfNotExists(stockAssetPath('hooks'), path.join(dir, 'hooks'));
+                copyIfNotExists(stockAssetPath('www'), path.join(dir, 'www'));
+                copyIfNotExists(stockAssetPath('hooks'), path.join(dir, 'hooks'));
```
Comment (brody2consult) [inline index.js]: I wanted to make it more clear about what this function actually does? Wonder if we should just remove the separate function altogether?

**Review by brody2consult — COMMENTED** [review]
I got a few more questions, looks great in general, I will try running spec on your changes next.

File: index.js (hunk 2 above)
Comment (raphinesse) [reply index.js]: I didn't check, but this case might not be covered or not sufficiently tested. Or I'm just plainly wrong :wink: 

But as far as I understand it, with `copySync` you *always* have the full file path in either argument. Whether you copy file or folder. This leads to less surprising if a bit wordier copying IMO.

File: index.js (hunk 4 above)
Comment (raphinesse) [reply index.js]: Well, the function would also work for plain files. But for that case the name makes no sense.

I'd keep it for now. It will have to be changed anyway in case some of apache/cordova-discuss#89 is accepted.
